lib/modules/us_core_guidance/clinicalnotes_sequence.rb

```diff
@@ -51,175 +41,166 @@ class USCoreR4ClinicalNotesSequence < SequenceBase
-          link 'https://www.hl7.org/fhir/us/core/clinical-notes-guidance.html'
-          description %(
-          )
-          versions :r4
-        end
-
-        test_clinical_notes_document_reference('http://loinc.org|34117-2')
+        parse_diagnostic_report_resources(resources, resource_class, patient_id)
       end
 
-      test :have_procedures_note do
-        metadata do
-          id '04'
-          name 'Server returns Procedures Note from DocumentReference search by patient+type'
-          link 'https://www.hl7.org/fhir/us/core/clinical-notes-guidance.html'
-          description %(
-          )
-          versions :r4
-        end
-
-        test_clinical_notes_document_reference('http://loinc.org|28570-0')
-      end
+      def parse_diagnostic_report_resources(resources, resource_class, patient_id)
+        category_found = []
+        attachments = {}
 
-      test :have_progress_note do
-        metadata do
-          id '05'
-          name 'Server returns Progress Note from DocumentReference search by patient+type'
-          link 'https://www.hl7.org/fhir/us/core/clinical-notes-guidance.html'
-          description %(
-          )
-          versions :r4
-        end
+        resources&.select { |r| r.resourceType == resource_class }&.each do |resource|
+          resource.category&.each do |category|
+            code = category.coding&.map { |coding| coding.code if CATEGORY_REQUIRED.include?(coding.code) }&.compact
 
-        test_clinical_notes_document_reference('http://loinc.org|11506-3')
-      end
+            if code.present?
+              category_found << code.first if category_found.exclude?(code.first)
 
-      test :have_cardiology_report do
-        metadata do
-          id '06'
-          name 'Server returns Cardiology report from DiagnosticReport search by patient+category'
-          link 'https://www.hl7.org/fhir/us/core/clinical-notes-guidance.html'
-          description %(
-          )
-          versions :r4
+              # Save DiagnosticReport.presentedForm.url for later test.
+              # Our current understanding is that Inferno only need to test the attachment for the three required DiagonistcReport
+              resource.presentedForm&.select { |attachment| !attachments.key?(attachment&.url) }&.each do |attachment|
+                attachments[attachment.url] = resource.id
+              end
+            end
+          end
         end
 
-        test_clinical_notes_diagnostic_report('http://loinc.org|LP29708-2')
+        report_attachments[patient_id] = attachments unless attachments.empty?
+
+        category_found
       end
 
-      test :have_pathology_report do
+      test :have_clinical_notes do
         metadata do
-          id '07'
-          name 'Server returns Pathology report from DiagnosticReport search by patient+category'
+          id '01'
+          name 'Server shall have minmum five types of DocumentReference and three categories of DiagnosticReport'
```

Comment:
Typo (minmum).  How about this:

```suggestion
          name 'Server demonstrates support for the required DocumentReference types and DiagnosticReport categories.'
```